sdk/go/common/resource/properties_path_test.go

```diff
@@ -144,6 +144,30 @@ func TestPropertyPath(t *testing.T) {
 		})
 	}
 
+	simpleCases := []struct {
+		path   string
+		parsed PropertyPath
+	}{
+		{
+			`root["*"].field[1]`,
```

Comment:
Feel there should be a `root[*].field[1]` test as well given that's what this is adding support for

Reply:
@pgavlin Which tests would you like to see added to `TestPropertyPathContains`? `TestPropertyPathContains` takes as input a `PropertyPath`. Since `foo[*].bar` and `foo["*"].bar` both parse into the same property path `PropertyPath{"foo", "*", "bar"}`, I don't think that my change introduces any new surface area for the `Contains` method.

Reply:
> `TestPropertyPathContains` takes as input a `PropertyPath`

Ah, so it does--I did not read the setup for that test closely enough. Nevermind, then!